Duplicate fields when UNIQUE constraint is used.

#3 PR brought new problem. Now fields with unique constraints are duplicated in some cases. 
<img width="708" alt="screen shot 2019-02-02 at 15 48 19" src="https://user-images.githubusercontent.com/1456389/52164935-270b9880-2702-11e9-8a67-3442b88138a7.png">

I'm trying to fix it but also want to cover this functionality with tests. 

Trying to run tests gives me:

```
test/model/users.go:4:2: cannot find package "github.com/dizzyfool/genna/test/geo" in any of:

	/usr/local/Cellar/go/1.9.2/libexec/src/github.com/dizzyfool/genna/test/geo (from $GOROOT)

	/Users/kuzma/Documents/Workspace/go/src/github.com/dizzyfool/genna/test/geo (from $GOPATH)

Process finished with exit code 1
```

--- a/database/store.go
+++ b/database/store.go
@@ -78,8 +78,8 @@ func (r *relationRow) Relation() model.Relation {
 	}
 }
 
-// Tables get basic tables information among selected schemas
-func (s *Store) Tables(schema []string) ([]model.Table, error) {
+func (s *Store) queryTables(schema []string) ([]columnRow, error) {
+	rows := make([]columnRow, 0)
 	query := `
 		with
 		    enums as (
@@ -101,19 +101,34 @@ func (s *Store) Tables(schema []string) ([]model.Table, error) {
 		        left join pg_namespace sch on sch.oid = tb.relnamespace
 		        left join pg_attribute col on col.attrelid = tb.oid
 		        where col.attndims > 0
+		    ),
+		    info as (
+				select distinct
+				 	kcu.table_schema as "table_schema",
+					kcu.table_name   as "table_name",
+					kcu.column_name  as "column_name",
+					array_agg((
+						select constraint_type 
+						from information_schema.table_constraints tc 
+						where tc.constraint_name = kcu.constraint_name 
+							and tc.constraint_schema = kcu.constraint_schema 
+							and tc.constraint_catalog = tc.constraint_catalog
+					)) as "constraint_types"
+				from information_schema.key_column_usage kcu
+				group by kcu.table_schema, kcu.table_name, kcu.column_name
 		    )
 		select distinct c."table_schema"                  as "schema",
 		                c."table_name"                    as "table",
 		                c."column_name"                   as "column",
 		                case
-		                when f.constraint_type is null
+		                when i.constraint_types is null
 		                then false
-		                else f.constraint_type = 'PRIMARY KEY'
-		                end                               as "pk",
-		                f.constraint_type = 'FOREIGN KEY' as "fk",
-		                c."is_nullable" = 'YES'           as "nullable",
-		                c."data_type" = 'ARRAY'           as "array",
-		                coalesce(a.array_dims, 0)         as "dims",
+		                else 'PRIMARY KEY'=ANY(i.constraint_types)
+		                end                                   as "pk",
+		                'FOREIGN KEY'=ANY(i.constraint_types) as "fk",
+		                c."is_nullable" = 'YES'               as "nullable",
+		                c."data_type" = 'ARRAY'               as "array",
+		                coalesce(a.array_dims, 0)             as "dims",
 		                case
 		                when e.is_enum = true
 		                then 'varchar'
@@ -122,18 +137,24 @@ func (s *Store) Tables(schema []string) ([]model.Table, error) {
 		                c.column_default                  as "default"
 		from information_schema.tables t
 		left join information_schema.columns c using (table_name, table_schema)
-		left join information_schema.key_column_usage k using (table_name, table_schema, column_name)
-		left join information_schema.table_constraints f using (table_name, table_schema, constraint_name)
+		left join info i using (table_name, table_schema, column_name)
 		left join arrays a using (table_name, table_schema, column_name)
 		left join enums e using (table_name, table_schema, column_name)
 		where t."table_schema" in (?)
 		  and t.table_type = 'BASE TABLE'
-		  and (f.constraint_type in ('PRIMARY KEY', 'FOREIGN KEY', 'UNIQUE') or f.constraint_type is null)
-		order by 1, 2, 4 desc nulls last;
+		order by 1, 2, 4 desc nulls last
 	`
 
-	rows := make([]columnRow, 0)
-	if _, err := s.db.Query(&rows, query, pg.In(schema)); err != nil {
+	_, err := s.db.Query(&rows, query, pg.In(schema))
+
+	return rows, err
+}
+
+// Tables get basic tables information among selected schemas
+func (s *Store) Tables(schema []string) ([]model.Table, error) {
+
+	rows, err := s.queryTables(schema)
+	if err != nil {
 		return nil, errors.Wrap(err, "getting table info error")
 	}
 

--- a/database/store_test.go
+++ b/database/store_test.go
@@ -1,0 +1,77 @@
+package database
+
+import (
+	"github.com/dizzyfool/genna/model"
+	"go.uber.org/zap"
+	"testing"
+)
+
+func TestLive(t *testing.T) {
+
+	config := zap.NewProductionConfig()
+	config.OutputPaths = []string{"stdout"}
+	config.Encoding = "console"
+	logger, _ := config.Build()
+	url := `postgres://genna:genna@localhost:5432/genna?sslmode=disable`
+	db, _ := NewDatabase(url, logger)
+
+	store := NewStore(db)
+
+	createTestTable(store)
+
+	var schemas []string
+	schemas = append(schemas, model.PublicSchema)
+	rows, _ := store.queryTables(schemas)
+
+	t.Run("Should be 5 rows", func(t *testing.T) {
+		got := len(rows)
+		if got != 5 {
+			t.Errorf("There should be 5 rows but I see %v", got)
+		}
+	})
+
+	t.Run("Should not duplicate columns", func(t *testing.T) {
+		if got := isThereDuplicates(rows); got != false {
+			t.Error("Duplicated columns detected")
+		}
+	})
+
+}
+
+// Create table with complex constraints for testing
+func createTestTable(store *Store) {
+	store.db.Model().Exec(`
+	drop table if exists contracts;
+	drop table if exists employees;
+
+	CREATE TABLE employees (
+		id serial NOT null,
+		CONSTRAINT employees_pk PRIMARY KEY (id)
+	);
+
+	CREATE TABLE contracts (
+		employee_id int4 NOT NULL,
+		email varchar NOT NULL,
+		id1 int NOT NULL,
+		id2 varchar NOT NULL,
+		CONSTRAINT contracts_un1 CHECK (id1 > 0),
+		CONSTRAINT contracts_un2 UNIQUE (email),
+		CONSTRAINT contracts_un3 UNIQUE (email, employee_id),
+		CONSTRAINT contracts_un4 UNIQUE (id1, id2),
+		CONSTRAINT contracts_employees_fk FOREIGN KEY (employee_id) REFERENCES employees(id)
+	);
+	`)
+}
+
+// Check if there are duplicated columns
+func isThereDuplicates(rows []columnRow) bool {
+	encountered := map[string]bool{}
+	for _, v := range rows {
+		if encountered[v.TblName+v.ColumnName] == true {
+			return true
+		} else {
+			encountered[v.TblName+v.ColumnName] = true
+		}
+	}
+	return false
+}

